2016/presentations/Python_for_Networking_Devices/text.md

```diff
@@ -0,0 +1,139 @@
+(for network engineers who are trying to figure out how to code ;) ) but don't provide any ease of use in terms of
+vendor specificity. You will still have to deal with the little differences the vendors bring, different login
+procedured or additional escape chars). You will take care of different lines of configuration or show commands you
+need to pass into paramiko or pexpect for each device or vendor, or with the varying support of netconf on each
+device for ncclient.
+
+##Specific Network Vendor Libraries
+
+Since Python became more and more popular amongst network engineers, network vendors started to provide their own
+libraries to facilitate device interaction - to mention a few: 
+
+* Arista's [pyeapi](https://github.com/arista-eosplus/pyeapi)
+* Cisco IOS-XR [pyiosxr](https://github.com/fooelisa/pyiosxr)
+* Juniper's [py-junos-eznc](https://github.com/Juniper/py-junos-eznc )
+
+Each of them supports their specific device with its specific capabilities. Juniper's py-junos-eznc  provides access
+via Netconf, whereas pyiosxr 'mimics' netconf support (which is not directly available) via SSH (pexpect). They ease
+config operations on the devices, such as replacing or merging configurations in one go, or . Each of them very
+specific to the vendors capabilities though. 
+
+##Multivendor Libraries
+
+Especially in a multivendor environment managing configurations takes on a whole different level. In addition to
+managing differences (and similarities) per device, for example a similar base configuration (syslog servers, NTP
+servers, dns servers) but different BGP neighbors on devices in different locations. Different access methods have to
+be taken into account, one via SSH the other via Netconf. Maintaining scripts with various libraries per vendor is a
+pain in the a**.
+
+Efforts to improve this  have been started, for example with [netmiko](https://github.com/ktbyers/netmiko) which is a
+wrapper around paramiko and provides easy SSH access for plenty of networking equipment. On the Netconf front, or to
+be more specific, access which supports transactions on the device, a project called
+[napalm](https://github.com/napalm-automation/napalm) has been started last year.
+
+Of course there are also standardization efforts, like open config or Netconf already, but as far as standards go
+interworkings between different vendors have been proven to take a lot of time and effort and not be on the market as
+quickly as the engineers would like.
+
+<!---
+##How to deal with your configuration management?
+
+Independent of the library you use to access your devices, configurations still need to be managed per device (and
+vendor). For a more programatic approach, we use templating 
+
+We use configuration templating and process enforcement tools (such as jinja2, ansibe) that are helpful in creating
+procedures for your work environment.
+--->
```

Comment:
What is this comment doing in here? Should it be un-commented or removed completely?